src/tests/unit/tests/injected/tab-stops-requirement-evaluator.test.ts

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { HTMLElementUtils } from 'common/html-element-utils';
+import { DefaultTabStopsRequirementEvaluator } from 'injected/tab-stops-requirement-evaluator';
+import { getUniqueSelector } from 'scanner/axe-utils';
+import { FocusableElement } from 'tabbable';
+import { IMock, It, Mock } from 'typemoq';
+
+describe('TabStopsRequirementEvaluator', () => {
+    let htmlElementUtilsMock: IMock<HTMLElementUtils>;
+    let generateSelectorMock: IMock<typeof getUniqueSelector>;
+    let testSubject: DefaultTabStopsRequirementEvaluator;
+
+    const tabStopElement1: HTMLElement = { tagName: 'element1', outerHTML: 'html1' } as HTMLElement;
+    const tabStopElement2: HTMLElement = { tagName: 'element2', outerHTML: 'html2' } as HTMLElement;
+
+    beforeEach(() => {
+        htmlElementUtilsMock = Mock.ofType(HTMLElementUtils);
+        generateSelectorMock = Mock.ofType<typeof getUniqueSelector>();
+        generateSelectorMock.setup(m => m(It.isAny())).returns(element => element.tagName);
+        testSubject = new DefaultTabStopsRequirementEvaluator(
+            htmlElementUtilsMock.object,
+            generateSelectorMock.object,
+        );
+    });
+
+    test('addKeyboardNavigationResults', () => {
+        const tabbableTabStops = [
+            tabStopElement1 as FocusableElement,
+            tabStopElement2 as FocusableElement,
+        ];
+        const incorrectTabStops = new Set<HTMLElement>([tabStopElement2]);
+        expect(
+            testSubject.getKeyboardNavigationResults(tabbableTabStops, incorrectTabStops),
+        ).toEqual([
+            {
+                description: 'Element element1 was expected, but not reached in tab order',
+                selector: ['element1'],
+                html: 'html1',
+            },
+        ]);
+
+        const correctTabStops = new Set<HTMLElement>([tabStopElement1, tabStopElement2]);
+        expect(testSubject.getKeyboardNavigationResults(tabbableTabStops, correctTabStops)).toEqual(
+            [],
+        );
+    });
+
+    test('addFocusOrderResults', () => {
+        htmlElementUtilsMock
```

Comment:
We like to break up tests by cases, so in this case the null and non-null scenarios will be separate.